library/mps/layer1.c

```diff
@@ -1160,6 +1160,7 @@ int mps_l1_fetch( mps_l1 *ctx, unsigned char **buf,
     MBEDTLS_MPS_ELSE_IF_DTLS( mode )
         return( l1_fetch_dgram( &ctx->raw.dgram.rd, buf, desired ) );
 #endif /* MBEDTLS_MPS_PROTO_DTLS */
+    RETURN( MBEDTLS_ERR_MPS_INTERNAL_ERROR );
```

Comment:
@lhuang04 I am puzzled by the compiler error. The helper macros `MBEDTLS_MPS_IF_TLS` and `MBEDTLS_ELSE_IF_DTLS` are defined in a way that the above will always unfold to precisely one block, or to `if() { ... } else if( 1 ) { ... }`, but never to `if() { ... } else if( nontrivial ) { ... }`, in which case the warning would be understandable.

Reply:
Ok, this is annoying, somehow `clang` complains about an `else if( mode == mode )` being trivial and wants it to be `else { ... }`. Which is functionally the same, but then the compiler complains about `mode` not being used, and you have to explicitly add something like`((void) mode)` which makes the code harder to read. The check `mode == mode` was just a transparent means to say `true` while pretending to use `mode`.

Reply:
Opened https://github.com/hannestschofenig/mbedtls/pull/156 to address this.